job-server/src/spark.jobserver/JobServer.scala

```diff
@@ -66,6 +75,44 @@ object JobServer {
 
   }
 
+  private def storeInitialJars(config: Config, jarManager: ActorRef): Unit = {
+    if(config.hasPath("spark.jobserver.job-jar-uris")) {
+      val initialJarsConfig = config.getConfig("spark.jobserver.job-jar-uris").root
+
+      logger.info("Adding initial job jars: {}", initialJarsConfig.render())
+
+      val initialJars =
+        initialJarsConfig
+          .asScala
+          .map { case (key, value) => (key, value.unwrapped.toString) }
+          .toMap
+
+      // Ensure that the jars exist
+      for(jarPath <- initialJars.values) {
+        val f = new java.io.File(jarPath)
```

Comment:
I think we can leave it, JobServer.scala is a bit hard to test since it's the outer app.